game-service/src/main/java/org/libertybikes/game/core/GameBoard.java

```diff
@@ -139,4 +144,60 @@ public boolean moveObjects() {
         return true;
     }
 
+    public boolean broadcastToAI() {
+        if (aiMap.isEmpty()) {
+            return false;
+        }
+
+        for (Player p : aiMap.keySet()) {
+            try {
+                AI ai = aiMap.get(p);
+                short[][] boardCopy = board.clone();
+                p.direction = (ai.processGameTick(boardCopy));
+            } catch (Exception e) {
+                System.out.println("Bot Exception: " + e.toString());
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     *
+     */
+    public void addAI() {
+        // Find first open player slot to fill, which determines position
+        short playerNum = -1;
+        for (short i = 0; i < takenPlayerSlots.length; i++) {
+            if (!takenPlayerSlots[i]) {
+                playerNum = i;
+                takenPlayerSlots[i] = true;
+                System.out.println("Player slot " + i + " taken");
+                break;
+            }
+        }
+
+        // Initialize Player
+        Player p = new Player("Hal", "Hal", playerNum);
```

Comment:
perhaps we should make the names unique, such as "Hal-1" for games where there are more than 1 AI players